### PR TITLE
docs: update docstrings for custom_exposure_query and custom_enrollments_query

### LINKED ISSUE #207: Clarify custom_enrollments_query docs
[These docs](https://mozilla.github.io/mozanalysis/api/experiment.html#mozanalysis.experiment.Experiment.get_single_window_data) are a bit confusing and incomplete for `custom_enrollments_query`:
```
A full SQL query to be used in the main query:

`WITH raw_enrollments AS ({custom_enrollments_query})`

N.B. this query’s results must be uniquely keyed by (client_id, branch), or else your results will be subtly wrong.
```
This text (or similar) is copied for multiple functions within this package, any updates should be made to all relevant function docs.

We should make the following updates to clarify:
- make it more clear that the `custom_enrollments_query` is only the part inside the parens (i.e., should _not_ include "WITH raw_enrollments AS"
- per #93, the query needs to produce columns client_id, branch, enrollment_date, and num_enrollment_events
- per #93, it should have no more than one row per (client_id, branch)

### PR DESCRIPTION
**Pull request by jaredsnyder**
Fixes #207

### CODE REVIEW

File: src/mozanalysis/experiment.py
Comment (mikewilli) [inline src/mozanalysis/experiment.py:929]: No f-strings in this file?

**Review by mikewilli — APPROVED** [review]
LGTM thanks! Surprised ruff doesn't care about the `.format()` calls in this file, but since this PR is about doc updates I think that's a problem for another time.

File: src/mozanalysis/experiment.py
Comment (jaredsnyder) [reply src/mozanalysis/experiment.py:929]: I'm still using the old ruff rules because I'm still waiting on merging the ruff branch.  Waiting for a final approval from one of the DS folks, I'll ask about it today

Comment (mikewilli) [reply src/mozanalysis/experiment.py:929]: Whoops :facepalm: that makes sense! Glad it's not another strange case of some files skipping some lint rules.

**Pull request merged**
